copy/copy.go

```diff
@@ -753,7 +751,9 @@ func computeDiffID(stream io.Reader, decompressor compression.DecompressorFunc)
 // and returns a complete blobInfo of the copied blob.
 func (c *copier) copyBlobFromStream(ctx context.Context, srcStream io.Reader, srcInfo types.BlobInfo,
 	getOriginalLayerCopyWriter func(decompressor compression.DecompressorFunc) io.Writer,
-	canModifyBlob bool, isConfig bool, bar *pb.ProgressBar) (types.BlobInfo, error) {
+	canModifyBlob bool, isConfig bool, bar *mpb.Bar) (types.BlobInfo, error) {
+	c.progressWG.Add(1)
+	defer c.progressWG.Done()
```

Comment:
Does `c.progressWG` do anything we need? The synchronization of parallel layer copy AFAICS happens independently via `copyGroup`, and the only effect of `mpb.WithWaitGroups` seems to be that `ic.c.progressPool.Wait()` blocks on `c.progressWG` in addition to doing everything else it does.

Reply:
That's intended as well. While testing this PR, we've experienced situations where the progress was still rendering although the copy operations have completed (including the one of the config). It's a background process with a specific refresh rate and the copier's WaitGroup makes sure that we progress only when it's done.

Reply:
> and the copier's WaitGroup makes sure that we progress only when it's done.

I can’t see how that works; the _only_ use of `progressWG` inside the code is `p.uwg.Wait()` immediately at the start of `Progress.Wait`, which AFAICS is immediately satisfied because `copyGroup.Wait` ensured that all layer copies are done, and `copyConfig` has returned.

`Progress.Wait` does wait for the separate render process to finish, via `pState.zeroWait` (“please exit”) and `Progress.done` (“exited”); I can’t see that this depends on `progressWG` in any way.